src/main/java/reposense/authorship/FileInfoExtractor.java

```diff
@@ -35,19 +52,101 @@
         // checks out to the latest commit of the date range to ensure the FileInfo generated correspond to the
         // git blame file analyze output
         GitChecker.checkoutToDate(config.getRepoRoot(), config.getBranch(), config.getUntilDate());
-        ArrayList<FileInfo> fileInfos = new ArrayList<>();
-        getAllFileInfo(config, Paths.get(config.getRepoRoot()), fileInfos);
+        List<FileInfo> fileInfos = new ArrayList<>();
+        String lastCommitHash = CommandRunner.getCommitHashBeforeDate(
+                config.getRepoRoot(), config.getBranch(), config.getSinceDate());
+
+        if (!lastCommitHash.isEmpty()) {
+            fileInfos = getEditedFileInfos(config, lastCommitHash);
+        } else {
+            getAllFileInfo(config, Paths.get(config.getRepoRoot()), fileInfos);
```

Comment:
Seems really weird that both `getAllFileInfo` and `getEditedFileInfos` are getters, but one has return type and the other don't.

Reply:
`getAllFileInfo` is a recursive function, which makes it slightly harder to have a return type (would have to keep appending all recursive called results).
Perhaps I should just make `getEditedFileInfos` not have a return type as well (pass in the `fileInfos` to be appended)?

Reply:
nvm, lets leave it for now